sdk/deploy.go

```diff
@@ -43,7 +48,43 @@ type DeployClient struct {
 	JSONClient *jsonclient.Client
 }
 
-// GetHistory interacts with Mia Platform APIs to retrieve a list of the lastest deploy.
+// DeployStrategy represents the type of deploy strategies that are available on the Console.
+type DeployStrategy string
+
+// DeployConfig is the details needed to trigger a deploy.
+type DeployConfig struct {
+	Environment         string
+	Revision            string
+	DeployAll           bool
+	ForceDeployNoSemVer bool
+}
+
+// DeployRequest is the body parameters needed to trigger a pipeline deploy.
+type DeployRequest struct {
+	Environment             string         `json:"environment"`
+	Revision                string         `json:"revision"`
+	DeployType              DeployStrategy `json:"deployType"`
+	ForceDeployWhenNoSemver bool           `json:"forceDeployWhenNoSemver"`
+}
+
+// DeployResponse is the response of the service after triggering a deploy pipeline.
+type DeployResponse struct {
+	Id  int    `json:"id"`
+	Url string `json:"url"`
+}
+
+// PipelineConfig represents a single triggered deploy pipeline in the config file.
+type PipelineConfig struct {
+	ProjectId   string `yaml:"projectid" mapstructure:"projectid"`
+	PipelineId  int    `yaml:"pipelineid" mapstructure:"pipelineid"`
+	Environment string `yaml:"environment" mapstructure:"environment"`
+}
+
+// PipelinesConfig represents a list in the config of all the pipelines
+// that have been triggered but not checked their status.
+type PipelinesConfig []PipelineConfig
```

Comment:
Why PipelinesConfig is set here? It should not be part of the cmd?

The sdk ignore the viper configuration to save

Reply:
Moved these types. In fact, the are not even used within the sdk.